packages/editor/src/components/block-list/style.scss

```diff
@@ -136,16 +136,6 @@
 	.editor-block-list__block:first-child .editor-block-list__block-edit {
 		margin-top: $block-padding - $block-spacing / 2;
 	}
-
-	// Space every block, and the default appender, using margins.
-	// This allows margins to collapse, which gives a better representation of how it looks on the frontend.
-	.editor-default-block-appender__content,
-	.editor-block-list__block .editor-block-list__block-edit,
```

Comment:
This is where we, currently in `master`, apply margins in a blanket statement to _all_ blocks.

Why should we _not_ do this? 

Because some blocks might want their own margins, or no margins, and shouldn't have to override. Besides, the theme might not have any margins registered, and by registering them on a per-block basis, the theme _will_ get these margins. This forces us to decide on a per block basis what, if any, margin is needed.

Reply:
>This forces us to decide on a per block basis what, if any, margin is needed.

But we apply a blanket rule to `[ class*="wp-block-" ] {` to apply those margins? Is that not contradicting?

Reply:
Yes, that's definitely contradicting.

The thing is, I think of the Gutenberg editor style as a _vanilla stylesheet_. Same as if you create a stock HTML page with no CSS styles, you get the _default browser stylesheet_, with Times New Roman, etc. 

Only, for margins, we currently set those on `.editor-block-list__block-edit`, as opposed to on the _contents of the block itself_. For example instead of setting a margin on a `p` tag, we set the margin on the `.editor-block-list__block-edit` container of the Paragraph block.

Why does this matter? Only one reason — if you mean to <a href="https://wordpress.org/gutenberg/handbook/extensibility/theme-support/#editor-styles">style the editor itself</a> to match the theme, it makes more sense to write:

```
body.gutenberg-editor-page p,
body.gutenberg-editor-page ul,
body.gutenberg-editor-page ol,
... {
    margin: 1em 0;
}
```

... than it does to write:

```
body.gutenberg-editor-page .editor-block-list__block-edit {
    margin: 1em 0;
}
```

Sure, it would be nicer even if themers could write, simply:

```
p,
ul,
ol,
... {
    margin: 1em 0;
}
```

But unless Shadow DOM support drastically improves or we can put the editing canvas in an iframe, I doubt we can enable that. So by moving the margins to the block contents itself, at least we're taking a half-step to make styling the editor simpler, even if it's still the same margin.

What do you think?

Reply:
I can support the goal, but as implemented, `[ class*="wp-block-" ] {` would still match exactly what we're targeting today with `.editor-block-list__block-edit`.

>Sure, it would be nicer even if themers could write, simply:

In some side-chatting with @mtias and @youknowriad some ideas have been bounced around here with how we might be able to achieve effectively this by either rewriting the CSS to prepend necessary prefixing, or iterate them automatically when rendered in the DOM via [`document.styleSheets`](https://developer.mozilla.org/en-US/docs/Web/API/DocumentOrShadowRoot/styleSheets). Not sure if it'll be a foolproof solution, but worth exploring. I think @youknowriad said he might look into it?

Reply:
I did say, and I'll give it a try, no updates at the moment though :(

Reply:
For what it's worth, you can never count on dynamic/SSR blocks to match a `[class*="wp-block-"]` selector. (Data point: none of our custom SSR blocks have `wp-block-` classes.)

Reply:
> I can support the goal, but as implemented, [ class*="wp-block-" ] { would still match exactly what we're targeting today with .editor-block-list__block-edit.

While I agree that selector is not ideal or probably even feasible, it doesn't exactly target the same. I'm not trying to split hairs here, but look at the markup:

<img width="479" alt="screen shot 2018-08-15 at 09 29 33" src="https://user-images.githubusercontent.com/1204802/44136674-e4e53954-a06d-11e8-99c0-c6d1c2655a80.png">

- `.editor-block-list__block` is the parent wrapper of it all
- `.editor-block-list__block-edit` is a direct descendant of that
- `.wp-block-quote` is a direct descendant of `.editor-block-list__block-edit`

Only the third item is the same on the front-end and the back-end. 

I realize that having the markup be the same in the frontend and the backend is likely not feasible in the near future, without Shadow DOM or things like that, but it is a baby step towrads at least styling the same bits of markup in the editor and the frontend. The margin is the same, sure, but the selector is different. 

This all speaks to a larger issue that came up recently when @iseulde <a href="https://github.com/WordPress/gutenberg/pull/7890#issuecomment-411310168">rewrote the table</a> to use nested blocks. Essentially I wasn't able to write CSS that would make our nested blocks editor markup behave and look like the table on the front-end. Not easily at least.

I'm not sure what the best approach is — possibly this branch isn't it. But to your point about making a build process that matches front-end selectors to backend elements, this would not address the challenge for a table with nested blocks, where the sequence of parent-to-child elements in the markup is important (at least pending `display: contents`).

Reply:
Here you go https://github.com/WordPress/gutenberg/pull/9008

Reply:
> For what it's worth, you can never count on dynamic/SSR blocks to match a `[class*="wp-block-"]` selector. (Data point: none of our custom SSR blocks use `wp-block-` classes.)

@chrisvanpatten What class do those blocks have then? How would it be possible to style them?